### PR TITLE
Fix some memory leaks

### PR DESCRIPTION
**Pull request by perost**

### CODE REVIEW

**Comment by perost** [comment]
@adeas31: This fixes some of the memory leaks I found while running OMEdit through valgrind a while back. This is achieved by using `std::unique_ptr` instead of raw owning pointers, which has a lot of benefits:
* Memory is automatically deleted when the pointer reaches end of life.
* No need to zero out the pointer.
* Makes it clear who actually owns the memory it points to.
* Forces you to implement a copy constructor/assignment operator if you want to make a copy of the class containing the pointer. Previously you could accidentally copy it via the default copy constructor, which is bad™ when the class contains raw owning pointers.

I know you don't usually use smart pointers in OMEdit, so please check if this looks good to you. I think it would be a good idea to use smart pointers more, because there are still a lot of memory leaks remaining. However, they're more complicated to fix due to ambiguity on who actually owns the memory, so I only fixed some of the trivial cases for now.

File: OMEdit/OMEditLIB/Element/Element.cpp [hunk 1]
```diff
@@ -2337,7 +2317,7 @@ void Element::createStateElement()
     extents << QPointF(-100, -100) << QPointF(100, 100);
     mpStateElementRectangle->setExtents(extents);
   } else {
-    mpStateElementRectangle = 0;
+    mpStateElementRectangle.release();
```
Comment (adeas31) [inline OMEdit/OMEditLIB/Element/Element.cpp:2320]: We should call `reset` here instead of `release`.

**Review by adeas31 — APPROVED** [review]
Looks good!

File: OMEdit/OMEditLIB/Element/Element.cpp (hunk 1 above)
Comment (perost) [reply OMEdit/OMEditLIB/Element/Element.cpp]: Yes, good catch.

**Pull request merged**